common/python/ax/cluster_management/argo_cluster_manager.py

```diff
@@ -160,34 +162,87 @@ def _set_env_if_present(self, args):
 
     def _get_s3_proxy_port(self, kubeconfig):
         k8s = KubernetesApiClient(config_file=kubeconfig)
-        resp = k8s.api.list_namespaced_service("default")
+        resp = k8s.api.list_namespaced_service(S3PROXY_NAMESPACE)
         for i in resp.items:
             if i.metadata.name == "s3proxy":
                 return i.spec.ports[0].node_port
-
         return None
 
     def _get_s3_proxy_endpoint(self, kubeconfig):
         host = get_host_ip(kubeconfig)
         port = self._get_s3_proxy_port(kubeconfig)
-        return "http://" + host + ":" + port
+        return "http://" + host + ":" + str(port)
+
+    def _install_s3_proxy(self, kube_config):
+        logger.info("Installing s3 proxy ...")
+        self._install_s3_proxy_namespace(kube_config)
+        self._install_s3_proxy_pvc(kube_config)
+        self._install_s3_proxy_service(kube_config)
+        self._install_s3_proxy_pod(kube_config)
+
+    def _install_s3_proxy_namespace(self,kube_config):
+        k8s = KubernetesApiClient(config_file=kube_config)
+        resp = k8s.api.list_namespace()
+        for i in resp.items:
+            if i.metadata.name == S3PROXY_NAMESPACE:
+                return None
+        subprocess.check_call(["kubectl", "--kubeconfig", kube_config, "create", "-f", "/ax/config/service/argo-wfe/axs3-namespace.yml"])
```

Comment:
Can you use swagger_client APIs for creating the kubernetes objects instead of kubectl?

Reply:
swagger_client api doesn't accept yaml